internal/catalog/internal/controllers/endpoints/controller.go

```diff
@@ -388,3 +392,28 @@ func workloadToEndpoint(svc *pbcatalog.Service, data *workloadData) *pbcatalog.E
 		Identity:     data.workload.Identity,
 	}
 }
+
+func workloadIdentityStatusFromEndpoints(endpoints *pbcatalog.ServiceEndpoints) *pbresource.Condition {
+	gatherIdentities := func(endpoints *pbcatalog.ServiceEndpoints) []string {
+		if endpoints == nil {
+			return nil
+		}
+
+		var identities []string
+		for _, ep := range endpoints.GetEndpoints() {
+			identities = append(identities, ep.GetIdentity())
+		}
```

Comment:
I think you should collect these in a `map[string]struct{}` and then use `maps.Keys` to dump the keys and sort those to remove all of the duplicates.

Reply:
done.